src/reconstruct_loopset.jl

```diff
@@ -25,12 +25,29 @@ function Loop(ls, l, ::Type{StaticUnitRange{L,U}}) where {L,U}
     Loop(gensym(:n), L, U, Symbol(""), Symbol(""), true, true)::Loop
 end
 
+function Loop(ls::LoopSet, l::Int, k::Int, ::Type{<:CartesianIndices{N}}) where N
+    start = gensym(:loopstart); stop = gensym(:loopstop)
+    axisexpr = Expr(:ref, Expr(:., Expr(:ref, :lb, l), QuoteNode(:indices)), k)
+    pushpreamble!(ls, Expr(:(=), start, Expr(:macrocall, Symbol("@inbounds"), LineNumberNode(@__LINE__, Symbol(@__FILE__)), Expr(:(.), axisexpr, QuoteNode(:start)))))
```

Comment:
Currently, the package assumes arrays have been adjusted to 0-indexing.
I.e., note the `staticm1` (static minus 1) here: https://github.com/chriselrod/LoopVectorization.jl/blob/master/src/graphs.jl#L359

I think it would be simpler to instead leave things [whatever they originally were] indexed, and only subtract 1 while lowering [here](https://github.com/chriselrod/LoopVectorization.jl/blob/master/src/graphs.jl#L62).

Meaning don't change this line of code, but either change the others to follow this assumption or wait for me to.